library/conn_bridge.c

```diff
@@ -32,36 +32,60 @@ struct ziti_bridge_s {
     bool ziti_eof;
     bool input_eof;
     ziti_connection conn;
-    uv_stream_t *input;
-    uv_stream_t *output;
+    uv_handle_t *input;
+    uv_handle_t *output;
     uv_close_cb close_cb;
     void *data;
     struct fd_bridge_s *fdbr;
     pool_t *input_pool;
     bool input_throttle;
+    unsigned long idle_timeout;
+    uv_timer_t *idle_timer;
 };
 
 static ssize_t on_ziti_data(ziti_connection conn, const uint8_t *data, ssize_t len);
 
 static void bridge_alloc(uv_handle_t *h, size_t req, uv_buf_t *b);
+static void close_bridge(struct ziti_bridge_s *br);
 
 static void on_input(uv_stream_t *s, ssize_t len, const uv_buf_t *b);
+static void on_udp_input(uv_udp_t *udp, ssize_t len, const uv_buf_t *b, const struct sockaddr *addr, unsigned int flags);
 
+extern int ziti_conn_bridge(ziti_connection conn, uv_handle_t *handle, uv_close_cb on_close) {
+    if (handle == NULL) return UV_EINVAL;
+
+    if ( !(handle->type == UV_TCP || handle->type == UV_NAMED_PIPE ||
+           handle->type == UV_TTY || handle->type == UV_UDP )) {
+        return UV_EINVAL;
```

Comment:
i feel like this would be a useful place for a log message telling the consumer what/why it's not working. i see it returns an error code, but it might be helpful. also this logic should be reflected in the doc

Reply:
valid handles are specified in the doc